beacon_node/src/config.rs

```diff
@@ -50,26 +54,44 @@ pub fn get_config<E: EthSpec>(
     client_config.set_data_dir(get_data_dir(cli_args));
 
     // If necessary, remove any existing database and configuration
-    if client_config.data_dir().exists() && cli_args.get_flag("purge-db") {
-        // Remove the chain_db.
-        let chain_db = client_config.get_db_path();
-        if chain_db.exists() {
-            fs::remove_dir_all(chain_db)
-                .map_err(|err| format!("Failed to remove chain_db: {}", err))?;
-        }
-
-        // Remove the freezer db.
-        let freezer_db = client_config.get_freezer_db_path();
-        if freezer_db.exists() {
-            fs::remove_dir_all(freezer_db)
-                .map_err(|err| format!("Failed to remove freezer_db: {}", err))?;
-        }
-
-        // Remove the blobs db.
-        let blobs_db = client_config.get_blobs_db_path();
-        if blobs_db.exists() {
-            fs::remove_dir_all(blobs_db)
-                .map_err(|err| format!("Failed to remove blobs_db: {}", err))?;
+    if client_config.data_dir().exists() {
+        if cli_args.get_flag("purge-db-force") {
+            let chain_db = client_config.get_db_path();
+            let freezer_db = client_config.get_freezer_db_path();
+            let blobs_db = client_config.get_blobs_db_path();
+            purge_db(chain_db, freezer_db, blobs_db)?;
+        } else if cli_args.get_flag("purge-db") {
+            if std::io::stdin().is_terminal() {
```

Comment:
`--purge-db` will now check to see if it running interactively before purging. If it isn't it will not prompt for confirmation, but instead log a warning and ignore the purge. This should prevent users running under systemd from accidentally stalling Lighthouse

Reply:
I think this check should maybe be `is_terminal() || stdin_inputs`. If we are reading from `stdin` then it doesn't matter if we are running interactively. E.g. someone could run `lighthouse bn --purge-db < confirm` in a script or daemon

Reply:
Added in [c2f1146](https://github.com/sigp/lighthouse/pull/6154/commits/c2f1146d5b17292a3da88c21b303a990e16398c6)

This now allows you to do `lighthouse bn --purge-db --stdin-inputs <<< confirm`

Note that this will mean that if Lighthouse is running under Windows, it will never ignore the flag, and will always block until it gets a response from stdin